extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php

```diff
@@ -296,4 +301,127 @@ public function testIsTypeSupportedInTemplatesDataProvider() {
 			]
 		];
 	}
+
+	/**
+	 * @param $mockParams
+	 * @param $isWikiaMobile
+	 * @param $result
+	 * @param $description
+	 * @dataProvider testGetAdjustedImageSizeDataProvider
+	 */
+	public function testGetAdjustedImageSize( $mockParams, $isWikiaMobile, $result, $description ) {
+		$mock = $this->getMockBuilder( 'Wikia\PortableInfobox\Helpers\PortableInfoboxRenderServiceHelper' )
+			->setMethods( [ 'isWikiaMobile' ] )
+			->getMock();
+		$mock->expects( $this->any() )->method( 'isWikiaMobile' )->will( $this->returnValue( $isWikiaMobile ) );
+
+		$file = $this->createWikiaFileHelperMock( $mockParams );
+
+		$this->assertEquals(
+			$result,
+			$mock->getAdjustedImageSize( $file ),
+			$description
+		);
+	}
+
+	public function testGetAdjustedImageSizeDataProvider() {
+		return [
+			[
+				'mockParams' => [
+					'fileHeight' => 2000,
+					'fileWidth' => 3000
+				],
+				'isWikiaMobile' => false,
+				'result' => [
+					'height' => 500,
+					'width' => 270
```

Comment:
Shouldn't this case generate:
```
[
					'height' => 180,
					'width' => 270
]
```
As far as I recall we were supposed to avoid cropping images / changing aspect ratio, unless I understand the test case wrongly